src/client/app/effects/github.ts

```diff
@@ -120,14 +127,21 @@ ${Strings.gistSharedDialogEnd}
 
             return gist;
         })
-        .mergeMap(gist => Observable.from([
-            new GitHub.LoadGistsAction(),
-            new GitHub.ShareSuccessAction(gist)
-        ]))
-        .catch(exception => Observable.from([
-            new UI.ReportErrorAction(Strings.gistShareFailed, exception),
-            new GitHub.ShareFailedAction()
-        ]));
+        .mergeMap((gist) => Observable.from([
+                new GitHub.LoadGistsAction(),
+                new GitHub.ShareSuccessAction(gist),
+                new Snippet.UpdateInfoAction({ id: storage.lastOpened.id, gist: gist.id, isOwned: true })])
+        )
+        .catch(exception => {
+            this._uiEffects.alert(
+                Strings.gistShareFailedBody + '\n\n' + Strings.reloadPrompt,
+                Strings.gistShareFailedTitle,
+                Strings.okButtonLabel)
+            .then(() => window.location.reload());
+            return Observable.from([
+                new GitHub.ShareFailedAction(),
+                new Snippet.UpdateInfoAction({ id: storage.lastOpened.id, gist: '', isOwned: false })]);
```

Comment:
Thanks for bringing this up. Have modified code to pass through the `id` from the `payload` up above. That should address concerns with using `storage` (which has now been removed) and the un-owning of the snippet (since the `isOwned` field has now also been removed).